src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -55,215 +71,294 @@ void apply() {
     }
 
     private void captureOs() {
-        Utils.sysProperty("os.name", providers).ifPresent(buildScan::tag);
+        sysProperty("os.name", providers).ifPresent(buildScan::tag);
     }
 
     private void captureIde() {
         if (!CiUtils.isCi(providers)) {
-            // Wait for projects to load to ensure Gradle project properties are initialized
-            gradle.projectsEvaluated(g -> {
-                Optional<String> ideaVendorName = Utils.sysProperty("idea.vendor.name", providers);
-                Optional<String> ideaVersion = Utils.sysProperty("idea.version", providers);
-                Optional<String> invokedFromAndroidStudio = Utils.projectProperty("android.injected.invoked.from.ide", providers, gradle);
-                Optional<String> androidStudioVersion = Utils.projectProperty("android.injected.studio.version", providers, gradle);
-                Optional<String> eclipseVersion = Utils.sysProperty("eclipse.buildId", providers);
-                Optional<String> ideaSync = Utils.sysProperty("idea.sync.active", providers);
-
-                if (ideaVendorName.isPresent()) {
-                    String ideaVendorNameValue = ideaVendorName.get();
-                    if (ideaVendorNameValue.equals("Google")) {
-                        // using androidStudioVersion instead of ideaVersion for compatibility reasons, those can be different (e.g. 2020.3.1 Patch 3 instead of 2020.3)
-                        tagIde("Android Studio", androidStudioVersion.orElse(""));
-                    } else if (ideaVendorNameValue.equals("JetBrains")) {
-                        tagIde("IntelliJ IDEA", ideaVersion.orElse(""));
-                    }
-                } else if (invokedFromAndroidStudio.isPresent()) {
-                    // this case should be handled by the ideaVendorName condition but keeping it for compatibility reason (ideaVendorName started with 2020.1)
-                    tagIde("Android Studio", androidStudioVersion.orElse(""));
-                } else if (ideaVersion.isPresent()) {
-                    // this case should be handled by the ideaVendorName condition but keeping it for compatibility reason (ideaVendorName started with 2020.1)
-                    tagIde("IntelliJ IDEA", ideaVersion.get());
-                } else if (eclipseVersion.isPresent()) {
-                    tagIde("Eclipse", eclipseVersion.get());
-                } else {
-                    buildScan.tag("Cmd Line");
-                }
-
-                if (ideaSync.isPresent()) {
-                    buildScan.tag("IDE sync");
-                }
-            });
+            // Prepare relevant properties for use at execution time
+            Map<String, Provider<String>> relevantProperties = new HashMap<>();
```

Comment:
can we name this variable `ideProperties`
```suggestion
            Map<String, Provider<String>> ideProperties = new HashMap<>();
```